src/ImageScaler.java

```diff
@@ -0,0 +1,169 @@
+                    minBottom = top;
+                    foundTop = true;
+                    top--;
+                    break;
+                }
+                x++;
+            }
+            top++;
+        }
+        boolean foundLeft = false;
+        while (left < minRight && !foundLeft) {
+            int y = height - 1;
+            while (y > top) {
+                if (writableRaster.getSample(left, y, 0) != 0) {
+                    minBottom = y;
+                    foundLeft = true;
+                    left--;
+                    break;
+                }
+                y--;
+            }
+            left++;
+        }
+        boolean foundBottom = false;
+        while (bottom > minBottom && !foundBottom) {
+            int x = width - 1;
+            while (x >= left) {
+                if (writableRaster.getSample(x, bottom, 0) != 0) {
+                    minRight = x;
+                    foundBottom = true;
+                    bottom++;
+                    break;
+                }
+                x--;
+            }
+            bottom--;
+        }
+
+        boolean foundRight = false;
+        while (right > minRight && !foundRight) {
+            int y = bottom;
+            while (y >= top) {
+                if (writableRaster.getSample(right, y, 0) != 0) {
+                    foundRight = true;
+                    right++;
+                    break;
+                }
+                y--;
+            }
+            right--;
+        }
+        return image.getSubimage(left, top, right - left + 1, bottom - top + 1);
+    }
+
+    public void scalImage(String unicode, String fontName) throws InvalidValueException {
+        if (unicode.equals("")) {
+            throw new InvalidValueException("Unicode can not be empty string");
+        } else if (fontName.equals("")) {
+            throw new InvalidValueException("Font can not be empty string");
+        }
+        BufferedImage imageBeforeScal = new BufferedImage(1, 1, BufferedImage.TYPE_4BYTE_ABGR);
+        Graphics2D graphic = imageBeforeScal.createGraphics();
+        Font font = null;
+        String regex = "([A-Z|a-z]:\\\\[^*|\"<>?\\n]*)|(\\\\\\\\.*?\\\\.*)";
+        if (Pattern.matches(regex, fontName)) {
+            try {
+                font = Font.createFont(Font.TRUETYPE_FONT, new File(fontName)).deriveFont(50f);
+                GraphicsEnvironment graphicsEnviroment = GraphicsEnvironment.getLocalGraphicsEnvironment();
+                graphicsEnviroment.registerFont(Font.createFont(Font.TRUETYPE_FONT, new File(fontName)));
+            } catch (IOException | FontFormatException e) {
+                System.out.println("Wrong font format!.");
+            }
+        } else {
+            font = new Font(fontName, Font.PLAIN, 50);
+        }
+
+        graphic.setFont(font);
+        graphic.dispose();
+
+        FontMetrics fontMetrics = graphic.getFontMetrics();
+        int width = fontMetrics.stringWidth(unicode);

```

Comment:
Warto tutaj dodać jakiś padding (np + 4 piksele). Bez tego często zdarza się, że litera się nie zmieści (polskie "ogonki"). Dodanie paddingu to najprostsze rozwiązanie tego problemu (wg mnie).